src/juliaexepath.ts

```diff
@@ -80,22 +79,27 @@ export async function getJuliaExePath() {
                     } else {
                         setNewJuliaExePath(p)
                     }
-                    foundJulia = true
                     break
                 }
                 catch (e) {
                 }
             }
-            if (!foundJulia) {
-                setNewJuliaExePath(getExecutablePath())
-            }
         }
         else {
             if (getExecutablePath().includes(path.sep)) {
                 setNewJuliaExePath(getExecutablePath().replace(/^~/, os.homedir()))
             } else {
                 // resolve full path
-                setNewJuliaExePath(await whichAsync(getExecutablePath()))
+                let fullPath: string | undefined = undefined
+                try {
+                    fullPath = await which(getExecutablePath())
+                }
+                catch (err) {
+                }
+
+                if (fullPath) {
+                    setNewJuliaExePath(fullPath)
+                }
```

Comment:
This is the really critical fix here: `which` throws an error if it doesn't find a program, and that error escaped into the `activate` function of the main extension.